dist/angular-breadcrumb.js

```diff
@@ -284,8 +284,95 @@ function BreadcrumbLastDirective($interpolate, $breadcrumb, $rootScope) {
+                post: function postLink(scope) {
+                    var labelWatchers = [];
+                    
+                    var registerWatchersText = function(labelWatcherArray, interpolationFunction, viewScope, step) {
+                        angular.forEach(getExpression(interpolationFunction), function(expression) {
+                            var watcher = viewScope.$watch(expression, function() {
+                                step.ncyBreadcrumbLabel = interpolationFunction(viewScope);
+                                renderLabel();
+                            });
+                            labelWatcherArray.push(watcher);
+                        });
+                    };
+
+                    var setupWatchers = function() {
+                        deregisterWatchers(labelWatchers);
+                        
+                        var viewScope = $breadcrumb.$getLastViewScope();
+                        var steps = $breadcrumb.getStatesChain();
+                        angular.forEach(steps, function (step) {
+                            if (step.ncyBreadcrumb && step.ncyBreadcrumb.label) {
+                                var parseLabel = $interpolate(step.ncyBreadcrumb.label);
+                                // Watcher for further viewScope updates
+                                registerWatchersText(labelWatchers, parseLabel, viewScope, step);
+                            }
+                        });
+                        
+                        renderLabel();
+                    };
+                    
+                    var renderLabel = function() {
+                        var viewScope = $breadcrumb.$getLastViewScope();
+                        var steps = $breadcrumb.getStatesChain();
+                        var combinedLabel = '';
+                        angular.forEach(steps, function (step) {
+                            if (step.ncyBreadcrumb && step.ncyBreadcrumb.label) {
+                                var parseLabel = $interpolate(step.ncyBreadcrumb.label);
+                                combinedLabel += separator + parseLabel(viewScope);
+                            } else {
+                                combinedLabel += separator + step.name;
+                            }
+                        });
+
+                        // strip leading separator
+                        if (combinedLabel.length > separator.length) {
+                            combinedLabel = combinedLabel.substring(separator.length);
+                        }
+
+                        scope.ncyBreadcrumbLabel = combinedLabel;
+                    };
+
+                    $rootScope.$on('$viewContentLoaded', function () {
+                        setupWatchers();
+                        renderLabel();
+                    });
+
+                    // View(s) may be already loaded while the directive's linking
+                    setupWatchers();
+                    renderLabel();
```

Comment:
Multiple performance-consuming instructions are duplicated in `setupWatchers` and `renderLabel`:
- Extract chain of states from the service
- Loop on the states
- Interpolation of labels

Couldn't we merge `setupWatchers` in `renderLabel`  ? (look at the ncyBreadcrumbLast directive)